Loop/View Controllers/StatusTableViewController.swift

```diff
@@ -1896,46 +1897,74 @@ extension StatusTableViewController: AddEditOverrideTableViewControllerDelegate
+        case .failure(let error):
+            log.default("Failure to setup pump manager with identifier '%{public}@': %{public}@", identifier, String(describing: error))
+        case .userInteractionRequired(var setupViewController):
+            setupViewController.pumpManagerCreateDelegate = self
+            setupViewController.pumpManagerOnboardDelegate = self
             setupViewController.completionDelegate = self
-            setupViewController.basalSchedule = deviceManager.loopManager.basalRateSchedule
-            setupViewController.maxBolusUnits = deviceManager.loopManager.settings.maximumBolus
-            setupViewController.maxBasalRateUnitsPerHour = deviceManager.loopManager.settings.maximumBasalRatePerHour
             show(setupViewController, sender: self)
-        } else {
-            let pumpManager = pumpManagerType.init(rawState: [:])
-            if let basalRateSchedule = deviceManager.loopManager.basalRateSchedule {
-                pumpManager?.syncBasalRateSchedule(items: basalRateSchedule.items, completion: { _ in })
-            }
-            deviceManager.pumpManager = pumpManager
+        case .success:
+            log.default("Pump manager with identifier '%{public}@' created and onboarded", identifier)
         }
     }
+}
 
-    func pumpManagerSetupViewController(_ pumpManagerSetupViewController: PumpManagerSetupViewController,
-                                        didSetUpPumpManager pumpManager: PumpManagerUI)
-    {
+extension StatusTableViewController: PumpManagerCreateDelegate {
+    func pumpManagerCreateNotifying(_ notifying: PumpManagerCreateNotifying, didCreatePumpManager pumpManager: PumpManagerUI) {
+        log.default("Pump manager with identifier '%{public}@' created", pumpManager.managerIdentifier)
         deviceManager.pumpManager = pumpManager
+    }
+}
 
-        if let basalRateSchedule = pumpManagerSetupViewController.basalSchedule {
+extension StatusTableViewController: PumpManagerOnboardDelegate {
+    func pumpManagerOnboardNotifying(_ notifying: PumpManagerOnboardNotifying, didOnboardPumpManager pumpManager: PumpManagerUI, withFinalSettings settings: PumpManagerSettings) {
+        precondition(pumpManager.isOnboarded)
+        log.default("Pump manager with identifier '%{public}@' onboarded", pumpManager.managerIdentifier)
+
+        if let basalRateSchedule = settings.basalSchedule {
             deviceManager.loopManager.basalRateSchedule = basalRateSchedule
         }
-
-        if let maxBasalRateUnitsPerHour = pumpManagerSetupViewController.maxBasalRateUnitsPerHour {
+        if let maxBasalRateUnitsPerHour = settings.maxBasalRateUnitsPerHour {
             deviceManager.loopManager.settings.maximumBasalRatePerHour = maxBasalRateUnitsPerHour
         }
-
-        if let maxBolusUnits = pumpManagerSetupViewController.maxBolusUnits {
+        if let maxBolusUnits = settings.maxBolusUnits {
```

Comment:
[nit] Should the onboarding delegate be the `deviceManager`?

Reply:
Yes, this will be addressed in https://tidepool.atlassian.net/browse/LOOP-2546. For right now I'm just following the current standard of having `StatusTableViewController` being a bit of a "dispatch coordinator". This will definitely need to change once onboarding appears *before* `StatusTableViewController` (and not on top of it, as it is now).